src/pipeline.rs

```diff
@@ -51,7 +71,14 @@ pub fn build_graphs(
 
     let mut line = String::new();
     while buffered.read_line(&mut line).unwrap() > 0 {
-        let split: Vec<&str> = line.trim().split('\t').collect();
+        let split: Vec<SmallVec<[String; SMALL_VECTOR_SIZE]>> = match &config.file_type {
```

Comment:
I think it's better not to do this with each new line. The type doesn't change while reading, so you can do it once outside of this loop.